### ISSUE TITLE
Cannot load round bales 125-150-180cm from Baling Contracts

### ISSUE THREAD

**Issue by bojanh66** [loki79uk/FS22_UniversalAutoload#100]
There is no option to load 125-150-180cm silage bales from Baling Contracts.
If i buy 125cm round silage bale from the shop, i can load it on the trailer, but cant do the same with bales from Contract missions.

I use version from Modhub...is this maybe fixed in the latest version from Github ?

**Comment by loki79uk** [comment]
Hi, thanks for the report.  It is worth trying with the latest version - I definitely added some features to copy settings for unknown mod bales/pallets with the same names, but I'm not sure why the contract bales would be different to standard bales.  Please try with 1.2.2.4 and let me know, but I will also set up a save game to test this over the weekend.

**Comment by bojanh66** [comment]
Hello, i have tried already last night and same problem...cannot load contract bales, there is no Shift + R option in F1 menu.
I have tried few other trailers from "Pallet Autoload Specialization" mod which is similar as your and it works with that mod, but i would prefer to use Universal Autoload.
If you could fix this small issue, i would appreciate it.
Thanks for your great work.

**Comment by bojanh66** [comment]
In the log file i have found this :
2022-09-24 17:47 *** UNIVERSAL AUTOLOAD - UNKNOWN OBJECT TYPE: FS22_LazyBales:roundbale125 ***
2022-09-24 17:47 *** UNIVERSAL AUTOLOAD - UNKNOWN OBJECT TYPE: FS22_LazyBales:roundbale180 ***

This was the problem, mod called LazyBales.
It is actually very useful mod on hilly maps, but unfortunately it doesnt work with Universal Autoload.
Square bales from Contract missions are supported, but round bales are not.

Now that i have found the problem, is there any possibility that you can make these bales to be supported ?

**Comment by loki79uk** [comment]
Hi, yes bales from that mod can easily be added to your local settings file.   I'm just setting up a game now with some contract bales to test with and then I'll try with that mod.  I'll post the settings that you need to use on here later today.

**Comment by loki79uk** [comment]
Courseplay is still doing the baling mission for me, but while we wait you could try adding this to your local settings file (inside the `<containerConfigurations>` tags:
```
	<containerConfiguration containerType="BALE" name="FS22_LazyBales:roundbale125" sizeX="1.300" sizeY="1.20" sizeZ="1.300" isRoundbale="true" flipYZ="true"/>
	<containerConfiguration containerType="BALE" name="FS22_LazyBales:roundbale150" sizeX="1.560" sizeY="1.20" sizeZ="1.560" isRoundbale="true" flipYZ="true"/>
	<containerConfiguration containerType="BALE" name="FS22_LazyBales:roundbale180" sizeX="1.835" sizeY="1.20" sizeZ="1.835" isRoundbale="true" flipYZ="true"/>
```

**Comment by loki79uk** [comment]
I tried to reproduce your bug with the LazyBales mod during a contract, but it must depend on the baler, because mine was creating standard round bales.  HOWEVER, I spawned some of them with EasyDev and can confirm that the settings in the previous comment will work.
[Here is a zipped settings file ](https://github.com/loki79uk/FS22_UniversalAutoload/files/9639496/UniversalAutoload.zip) that you need to unzip (GitHub won't allow uploaded xml files)
and then just drop into your mod settings folder: `%USERPROFILE%\Documents\My Games\FarmingSimulator2022\modSettings`

**Comment by bojanh66** [comment]
Thank you for your help, i can confirm it finally works.
Will you be able to update mod to work with Lazy Bales in the future ?

**Comment by loki79uk** [comment]
> Thank you for your help, i can confirm it finally works. Will you be able to update mod to work with Lazy Bales in the future ?

I'll have a think about this.  In the very early versions of UAL this mod would have worked with UAL, because I used only the i3d file name to 'identify' a pallet or bale.  That worked nicely when modders re-used the base game i3d files, as the stacking dimensions were still known and correct.  HOWEVER, the problem was that quite a few mods copied base game i3d files, but then edited the dimensions without changing the file name.  So I had to introduce a method to differentiate between pallets/bales with the same i3d file names, but different dimensions.

TLDR: currently any mods that use unchanged base game i3d files will not load with UAL (like LazyBales).  The alternative would be that mods with matching i3d file names but different dimensions would load in an unpredicatble way (either too far apart or overlapping).
